<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PreTSA">
<title>Model temporal gene expression patterns • PreTSA</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Model temporal gene expression patterns">
<meta property="og:description" content="PreTSA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PreTSA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/PreTSA_spatial.html">Model spatial gene expression patterns</a>
    <a class="dropdown-item" href="../articles/PreTSA_temporal.html">Model temporal gene expression patterns</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/haotian-zhuang/PreTSA/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Model temporal gene expression patterns</h1>
                        <h4 data-toc-skip class="author">Haotian
Zhuang</h4>
            
            <h4 data-toc-skip class="date">February 01, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/haotian-zhuang/PreTSA/blob/HEAD/vignettes/PreTSA_temporal.Rmd" class="external-link"><code>vignettes/PreTSA_temporal.Rmd</code></a></small>
      <div class="d-none name"><code>PreTSA_temporal.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We will demonstrate how to use PreTSA to fit the gene expression
along pseudotime values and identify temporally variable genes
(TVGs).</p>
<p>The 3K peripheral blood mononuclear cells (PBMC3K) dataset was
downloaded from the <a href="https://www.10xgenomics.com/resources/datasets/3-k-pbm-cs-from-a-healthy-donor-1-standard-1-1-0" class="external-link">10x
website</a>. The processed PBMC3K dataset was downloaded using the R
package <code>SeuratData</code>. <code>TSCAN</code> was used to
construct a pseudotime trajectory from naive CD4 T cells to memory CD4 T
cells with the top 10 principal components (PCs) and the cell clusters
obtained from the dataset. Genes that are expressed in fewer than 5
cells within the trajectory were filtered out. We analyzed a final set
of 10,509 genes on 1,180 cells.</p>
</div>
<div class="section level2">
<h2 id="load-packages-and-datasets">Load packages and datasets<a class="anchor" aria-label="anchor" href="#load-packages-and-datasets"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/haotian-zhuang/PreTSA" class="external-link">PreTSA</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pbmc3k"</span>, <span class="st">"pbmc3k_cleaned.Rdata"</span>, package <span class="op">=</span> <span class="st">"PreTSA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="co">#&gt; Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   ..@ i       : int [1:953387] 22 52 157 168 190 252 350 430 486 554 ...</span></span>
<span><span class="co">#&gt;   ..@ p       : int [1:1181] 0 238 508 958 1290 1691 2211 2596 2842 3257 ...</span></span>
<span><span class="co">#&gt;   ..@ Dim     : int [1:2] 10509 1180</span></span>
<span><span class="co">#&gt;   ..@ Dimnames:List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:10509] "LINC00115" "NOC2L" "HES4" "ISG15" ...</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:1180] "CTAATAGAGCTATG" "TAGTCTTGGCTGTA" "ACTTGTACCTGTCC" "CACCCATGTTCTGT" ...</span></span>
<span><span class="co">#&gt;   ..@ x       : num [1:953387] 2.87 2.87 5.46 2.87 3.54 ...</span></span>
<span><span class="co">#&gt;   ..@ factors : list()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">pseudotime</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named int [1:1180] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr [1:1180] "CTAATAGAGCTATG" "TAGTCTTGGCTGTA" "ACTTGTACCTGTCC" "CACCCATGTTCTGT" ...</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-the-gene-expression-along-pseudotime-values">Fit the gene expression along pseudotime values<a class="anchor" aria-label="anchor" href="#fit-the-gene-expression-along-pseudotime-values"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/temporalFit.html">temporalFit</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, pseudotime <span class="op">=</span> <span class="va">pseudotime</span>, knot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">fitRes</span><span class="op">)</span></span>
<span><span class="co">#&gt;  num [1:10509, 1:1180] 0.00238 0.16975 0.00847 0.34266 -0.00666 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:10509] "LINC00115" "NOC2L" "HES4" "ISG15" ...</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:1180] "CTAATAGAGCTATG" "TAGTCTTGGCTGTA" "ACTTGTACCTGTCC" "CACCCATGTTCTGT" ...</span></span></code></pre></div>
<p>The argument “knot” indicates whether to select the optimal number of
knots automatically (FALSE by default). The argument “maxknotallowed” is
the user-defined maximum number of knots (10 by default).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/temporalFit.html">temporalFit</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, pseudotime <span class="op">=</span> <span class="va">pseudotime</span>, knot <span class="op">=</span> <span class="cn">T</span>, maxknotallowed <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="identify-temporally-variable-genes-tvgs">Identify temporally variable genes (TVGs)<a class="anchor" aria-label="anchor" href="#identify-temporally-variable-genes-tvgs"></a>
</h2>
<p>To account for the pseudotime uncertainty, we apply the similar
strategy used in <code>PseudotimeDE</code>. The argument
“pseudotime_permute” is a list of permuted pseudotime values from
subsampled cells. Each element in the list has the same format of the
argument “pseudotime”.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pseudotime_permute</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pbmc3k"</span>, <span class="st">"pbmc3k_pseudotime_permute.rds"</span>, package <span class="op">=</span> <span class="st">"PreTSA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pseudotime_permute</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 100</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">pseudotime_permute</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named int [1:944] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr [1:944] "AAATTCGAGGAGTG" "GAGATGCTGAATGA" "TATCTTCTAAACAG" "GCAAGACTACTGGT" ...</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/temporalTest.html">temporalTest</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, pseudotime <span class="op">=</span> <span class="va">pseudotime</span>, pseudotime_permute <span class="op">=</span> <span class="va">pseudotime_permute</span>, knot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>To reduce the computational time, users can ignore the pseudotime
uncertainty, which is the default setting.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/temporalTest.html">temporalTest</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, pseudotime <span class="op">=</span> <span class="va">pseudotime</span>, pseudotime_permute <span class="op">=</span> <span class="cn">NULL</span>, knot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>It returns a data frame with the p-value, FDR, and the number of
knots selected for each gene.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">testRes</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    fdr         pval knotnum</span></span>
<span><span class="co">#&gt; LINC00115 9.784399e-01 9.589589e-01       0</span></span>
<span><span class="co">#&gt; NOC2L     8.891055e-01 7.825888e-01       0</span></span>
<span><span class="co">#&gt; HES4      9.999038e-01 9.998087e-01       0</span></span>
<span><span class="co">#&gt; ISG15     2.309152e-15 1.098654e-17       0</span></span>
<span><span class="co">#&gt; C1orf159  2.306248e-01 5.202783e-02       0</span></span>
<span><span class="co">#&gt; TNFRSF18  1.130205e-06 2.097155e-08       0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-the-fitted-expression">Visualize the fitted expression<a class="anchor" aria-label="anchor" href="#visualize-the-fitted-expression"></a>
</h2>
<p>As an example, <em>IL32</em> is an known gene associated to T cell
activation process. The fitted PreTSA curve shows an increasing
trend.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">genename</span> <span class="op">&lt;-</span> <span class="st">"IL32"</span></span>
<span><span class="va">df.gene</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ptime <span class="op">=</span> <span class="va">pseudotime</span>, ori <span class="op">=</span> <span class="va">expr</span><span class="op">[</span><span class="va">genename</span>, <span class="op">]</span>, pretsa <span class="op">=</span> <span class="va">fitRes</span><span class="op">[</span><span class="va">genename</span>, <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">df.gene</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ptime</span>, y <span class="op">=</span> <span class="va">ori</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"orange"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ptime</span>, y <span class="op">=</span> <span class="va">pretsa</span><span class="op">)</span>, linewidth <span class="op">=</span> <span class="fl">2</span>, color <span class="op">=</span> <span class="st">"royalblue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Pseudotime"</span>, y <span class="op">=</span> <span class="st">"Expression"</span>, title <span class="op">=</span> <span class="va">genename</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span>, face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="PreTSA_temporal_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<details><summary><strong>Session Info</strong>
</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.4.2 Matrix_1.5-1  PreTSA_1.0   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] highr_0.10         pillar_1.8.1       bslib_0.4.2        compiler_4.2.1    </span></span>
<span><span class="co">#&gt;  [5] jquerylib_0.1.4    tools_4.2.1        digest_0.6.31      tibble_3.2.1      </span></span>
<span><span class="co">#&gt;  [9] jsonlite_1.8.4     evaluate_0.19      memoise_2.0.1      lifecycle_1.0.3   </span></span>
<span><span class="co">#&gt; [13] gtable_0.3.1       lattice_0.20-45    pkgconfig_2.0.3    rlang_1.1.0       </span></span>
<span><span class="co">#&gt; [17] cli_3.6.1          rstudioapi_0.14    yaml_2.3.6         pkgdown_2.0.7     </span></span>
<span><span class="co">#&gt; [21] xfun_0.39          fastmap_1.1.0      withr_2.5.0        dplyr_1.1.1       </span></span>
<span><span class="co">#&gt; [25] stringr_1.5.0      knitr_1.41         generics_0.1.3     desc_1.4.2        </span></span>
<span><span class="co">#&gt; [29] fs_1.5.2           vctrs_0.6.1        sass_0.4.4         systemfonts_1.0.4 </span></span>
<span><span class="co">#&gt; [33] tidyselect_1.2.0   rprojroot_2.0.3    grid_4.2.1         glue_1.6.2        </span></span>
<span><span class="co">#&gt; [37] R6_2.5.1           textshaping_0.3.6  fansi_1.0.3        rmarkdown_2.19    </span></span>
<span><span class="co">#&gt; [41] farver_2.1.1       purrr_1.0.1        magrittr_2.0.3     scales_1.2.1      </span></span>
<span><span class="co">#&gt; [45] htmltools_0.5.4    matrixStats_0.63.0 splines_4.2.1      colorspace_2.0-3  </span></span>
<span><span class="co">#&gt; [49] labeling_0.4.2     ragg_1.2.5         utf8_1.2.2         stringi_1.7.8     </span></span>
<span><span class="co">#&gt; [53] munsell_0.5.0      cachem_1.0.6</span></span></code></pre></div>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Haotian Zhuang, Zhicheng Ji.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
