<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PreTSA">
<title>Model spatial gene expression patterns • PreTSA</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Model spatial gene expression patterns">
<meta property="og:description" content="PreTSA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PreTSA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/PreTSA_spatial.html">Model spatial gene expression patterns</a>
    <a class="dropdown-item" href="../articles/PreTSA_temporal.html">Model temporal gene expression patterns</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/haotian-zhuang/PreTSA/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Model spatial gene expression patterns</h1>
                        <h4 data-toc-skip class="author">Haotian
Zhuang</h4>
            
            <h4 data-toc-skip class="date">February 01, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/haotian-zhuang/PreTSA/blob/HEAD/vignettes/PreTSA_spatial.Rmd" class="external-link"><code>vignettes/PreTSA_spatial.Rmd</code></a></small>
      <div class="d-none name"><code>PreTSA_spatial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We will demonstrate how to use PreTSA to fit the gene expression
along spatial locations and identify spatially variable genes
(SVGs).</p>
<p>The Visium dataset of a human heart tissue was downloaded from the <a href="https://www.10xgenomics.com/resources/datasets/human-heart-1-standard-1-1-0" class="external-link">10x
website</a>. Mitochondrial genes and genes that are expressed in fewer
than 50 spots were filtered out. We analyzed a final set of 11,953 genes
on 4,247 spots. <code>SCTransform</code> with default settings was used
to normalize the raw count data.</p>
</div>
<div class="section level2">
<h2 id="load-packages-and-datasets">Load packages and datasets<a class="anchor" aria-label="anchor" href="#load-packages-and-datasets"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/haotian-zhuang/PreTSA" class="external-link">PreTSA</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"heart"</span>, <span class="st">"heart_cleaned.Rdata"</span>, package <span class="op">=</span> <span class="st">"PreTSA"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="co">#&gt; Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   ..@ i       : int [1:8104687] 9 10 16 22 23 45 50 59 61 68 ...</span></span>
<span><span class="co">#&gt;   ..@ p       : int [1:4248] 0 1970 4194 6523 8020 9234 11174 12758 15297 17812 ...</span></span>
<span><span class="co">#&gt;   ..@ Dim     : int [1:2] 11953 4247</span></span>
<span><span class="co">#&gt;   ..@ Dimnames:List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:11953] "LINC01409" "LINC01128" "NOC2L" "PERM1" ...</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:4247] "AAACAAGTATCTCCCA-1" "AAACACCAATAACTGC-1" "AAACAGAGCGACTCCT-1" "AAACAGCTTTCAGAAG-1" ...</span></span>
<span><span class="co">#&gt;   ..@ x       : num [1:8104687] 0.693 0.693 0.693 1.792 0.693 ...</span></span>
<span><span class="co">#&gt;   ..@ factors : list()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span>
<span><span class="co">#&gt;  int [1:4247, 1:2] 50 59 14 43 47 62 61 3 45 54 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:4247] "AAACAAGTATCTCCCA-1" "AAACACCAATAACTGC-1" "AAACAGAGCGACTCCT-1" "AAACAGCTTTCAGAAG-1" ...</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:2] "row" "col"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-the-gene-expression-along-spatial-locations">Fit the gene expression along spatial locations<a class="anchor" aria-label="anchor" href="#fit-the-gene-expression-along-spatial-locations"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialFit.html">spatialFit</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, coord <span class="op">=</span> <span class="va">coord</span>, knot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">fitRes</span><span class="op">)</span></span>
<span><span class="co">#&gt;  num [1:11953, 1:4247] 0.0242 0.1786 0.1103 0.2397 0.2105 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:11953] "LINC01409" "LINC01128" "NOC2L" "PERM1" ...</span></span>
<span><span class="co">#&gt;   ..$ : chr [1:4247] "AAACAAGTATCTCCCA-1" "AAACACCAATAACTGC-1" "AAACAGAGCGACTCCT-1" "AAACAGCTTTCAGAAG-1" ...</span></span></code></pre></div>
<p>The argument “knot” indicates whether to select the optimal number of
knots automatically (FALSE by default). The argument “maxknotallowed” is
the user-defined maximum number of knots (5 by default).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatialFit.html">spatialFit</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, coord <span class="op">=</span> <span class="va">coord</span>, knot <span class="op">=</span> <span class="cn">T</span>, maxknotallowed <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="identify-spatially-variable-genes-svgs">Identify spatially variable genes (SVGs)<a class="anchor" aria-label="anchor" href="#identify-spatially-variable-genes-svgs"></a>
</h2>
<p>It returns a data frame with the p-value, FDR, and the number of
knots selected for each gene.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialTest.html">spatialTest</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr</span>, coord <span class="op">=</span> <span class="va">coord</span>, knot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">testRes</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    fdr         pval knotnum</span></span>
<span><span class="co">#&gt; LINC01409 3.707785e-01 2.210843e-01       0</span></span>
<span><span class="co">#&gt; LINC01128 1.314814e-01 4.623516e-02       0</span></span>
<span><span class="co">#&gt; NOC2L     4.960952e-01 3.468474e-01       0</span></span>
<span><span class="co">#&gt; PERM1     2.940015e-01 1.544033e-01       0</span></span>
<span><span class="co">#&gt; HES4      6.518492e-02 1.686202e-02       0</span></span>
<span><span class="co">#&gt; ISG15     1.017506e-29 3.405022e-32       0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-the-fitted-expression">Visualize the fitted expression<a class="anchor" aria-label="anchor" href="#visualize-the-fitted-expression"></a>
</h2>
<p>The fitted PreTSA expression can be seamlessly incorporated into the
<code>Seurat</code> pipeline.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Attaching SeuratObject</span></span>
<span><span class="va">heart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Load10X_Spatial.html" class="external-link">Load10X_Spatial</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"heart"</span>, package <span class="op">=</span> <span class="st">"PreTSA"</span><span class="op">)</span>, filename <span class="op">=</span> <span class="st">"V1_Human_Heart_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">heart</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Spatial"</span></span>
<span><span class="va">heart</span><span class="op">[[</span><span class="st">"Fitted"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">fitRes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">heart</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Fitted"</span></span></code></pre></div>
<p>As an example, the fitted PreTSA expression of <em>MYH7</em> shows
that spots in the right side have significantly lower expression levels
than those in the left side.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genename</span> <span class="op">&lt;-</span> <span class="st">"MYH7"</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">heart</span>, features <span class="op">=</span> <span class="va">genename</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span>, legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="PreTSA_spatial_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<details><summary><strong>Session Info</strong>
</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] SeuratObject_4.1.3 Seurat_4.3.0       ggplot2_3.4.2      Matrix_1.5-1      </span></span>
<span><span class="co">#&gt; [5] PreTSA_1.0        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] Rtsne_0.16             colorspace_2.0-3       deldir_1.0-6          </span></span>
<span><span class="co">#&gt;   [4] ellipsis_0.3.2         ggridges_0.5.4         rprojroot_2.0.3       </span></span>
<span><span class="co">#&gt;   [7] fs_1.5.2               spatstat.data_3.0-0    rstudioapi_0.14       </span></span>
<span><span class="co">#&gt;  [10] farver_2.1.1           leiden_0.4.3           listenv_0.9.0         </span></span>
<span><span class="co">#&gt;  [13] bit64_4.0.5            ggrepel_0.9.2          fansi_1.0.3           </span></span>
<span><span class="co">#&gt;  [16] codetools_0.2-18       splines_4.2.1          cachem_1.0.6          </span></span>
<span><span class="co">#&gt;  [19] knitr_1.41             polyclip_1.10-4        jsonlite_1.8.4        </span></span>
<span><span class="co">#&gt;  [22] ica_1.0-3              cluster_2.1.3          png_0.1-8             </span></span>
<span><span class="co">#&gt;  [25] uwot_0.1.14            spatstat.sparse_3.0-0  shiny_1.7.4           </span></span>
<span><span class="co">#&gt;  [28] sctransform_0.3.5      compiler_4.2.1         httr_1.4.5            </span></span>
<span><span class="co">#&gt;  [31] fastmap_1.1.0          lazyeval_0.2.2         cli_3.6.1             </span></span>
<span><span class="co">#&gt;  [34] later_1.3.0            htmltools_0.5.4        tools_4.2.1           </span></span>
<span><span class="co">#&gt;  [37] igraph_1.3.5           gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">#&gt;  [40] RANN_2.6.1             reshape2_1.4.4         dplyr_1.1.1           </span></span>
<span><span class="co">#&gt;  [43] Rcpp_1.0.10            scattermore_0.8        jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt;  [46] pkgdown_2.0.7          vctrs_0.6.1            nlme_3.1-157          </span></span>
<span><span class="co">#&gt;  [49] spatstat.explore_3.0-5 progressr_0.12.0       lmtest_0.9-40         </span></span>
<span><span class="co">#&gt;  [52] spatstat.random_3.0-1  xfun_0.39              stringr_1.5.0         </span></span>
<span><span class="co">#&gt;  [55] globals_0.16.2         mime_0.12              miniUI_0.1.1.1        </span></span>
<span><span class="co">#&gt;  [58] lifecycle_1.0.3        irlba_2.3.5.1          goftest_1.2-3         </span></span>
<span><span class="co">#&gt;  [61] future_1.30.0          MASS_7.3-57            zoo_1.8-11            </span></span>
<span><span class="co">#&gt;  [64] scales_1.2.1           spatstat.utils_3.0-1   ragg_1.2.5            </span></span>
<span><span class="co">#&gt;  [67] promises_1.2.0.1       parallel_4.2.1         RColorBrewer_1.1-3    </span></span>
<span><span class="co">#&gt;  [70] yaml_2.3.6             memoise_2.0.1          reticulate_1.28       </span></span>
<span><span class="co">#&gt;  [73] pbapply_1.6-0          gridExtra_2.3          sass_0.4.4            </span></span>
<span><span class="co">#&gt;  [76] stringi_1.7.8          highr_0.10             desc_1.4.2            </span></span>
<span><span class="co">#&gt;  [79] rlang_1.1.0            pkgconfig_2.0.3        systemfonts_1.0.4     </span></span>
<span><span class="co">#&gt;  [82] matrixStats_0.63.0     evaluate_0.19          lattice_0.20-45       </span></span>
<span><span class="co">#&gt;  [85] tensor_1.5             ROCR_1.0-11            purrr_1.0.1           </span></span>
<span><span class="co">#&gt;  [88] labeling_0.4.2         patchwork_1.1.2        htmlwidgets_1.6.1     </span></span>
<span><span class="co">#&gt;  [91] bit_4.0.5              cowplot_1.1.1          tidyselect_1.2.0      </span></span>
<span><span class="co">#&gt;  [94] parallelly_1.33.0      RcppAnnoy_0.0.20       plyr_1.8.8            </span></span>
<span><span class="co">#&gt;  [97] magrittr_2.0.3         R6_2.5.1               generics_0.1.3        </span></span>
<span><span class="co">#&gt; [100] DBI_1.1.3              pillar_1.8.1           withr_2.5.0           </span></span>
<span><span class="co">#&gt; [103] fitdistrplus_1.1-8     abind_1.4-5            survival_3.3-1        </span></span>
<span><span class="co">#&gt; [106] sp_1.5-1               tibble_3.2.1           future.apply_1.10.0   </span></span>
<span><span class="co">#&gt; [109] hdf5r_1.3.7            KernSmooth_2.23-20     utf8_1.2.2            </span></span>
<span><span class="co">#&gt; [112] spatstat.geom_3.0-3    plotly_4.10.1          rmarkdown_2.19        </span></span>
<span><span class="co">#&gt; [115] grid_4.2.1             data.table_1.14.6      digest_0.6.31         </span></span>
<span><span class="co">#&gt; [118] xtable_1.8-4           tidyr_1.3.0            httpuv_1.6.7          </span></span>
<span><span class="co">#&gt; [121] textshaping_0.3.6      munsell_0.5.0          viridisLite_0.4.1     </span></span>
<span><span class="co">#&gt; [124] bslib_0.4.2</span></span></code></pre></div>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Haotian Zhuang, Zhicheng Ji.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
